modin/core/execution/dask/common/utils.py

```diff
@@ -39,6 +40,18 @@ def initialize_dask():
         memory_limit = Memory.get()
         worker_memory_limit = memory_limit // num_cpus if memory_limit else "auto"
         client = Client(n_workers=num_cpus, memory_limit=worker_memory_limit)
+        if ModinGithubCI.get():
+            # set these keys to run tests that write to the mock s3 service. this seems
+            # to be the way to pass environment variables to the workers:
+            # https://jacobtomlinson.dev/posts/2021/bio-for-2021/
+            client.run(
+                lambda: os.environ.update(
+                    {
+                        "AWS_ACCESS_KEY_ID": "foobar_key",
+                        "AWS_SECRET_ACCESS_KEY": "foobar_secret",
+                    }
+                )
```

Comment:
```suggestion
                lambda key_id=os.environ["AWS_ACCESS_KEY_ID"], access_key=os.environ["AWS_SECRET_ACCESS_KEY"]: os.environ.update(
                    {
                        "AWS_ACCESS_KEY_ID": key_id,
                        "AWS_SECRET_ACCESS_KEY": access_key,
                    }
                )
```
(not formatted)

General idea is to transfer whatever values were set on the head node.
Also I wonder if we should actually do this unconditionally, but with something like `os.environ.setdefault()` instead?